app/src/main/java/com/amaze/filemanager/filesystem/HybridFile.java

```diff
@@ -786,29 +723,14 @@ public Void execute(SFTPClient client) {
                   try {
                     for (RemoteResourceInfo info :
                         client.ls(SshClientUtils.extractRemotePathFrom(path))) {
-                      boolean isDirectory = info.isDirectory();
-                      if (info.getAttributes().getType().equals(FileMode.Type.SYMLINK)) {
-                        try {
-                          FileAttributes symlinkAttrs = client.stat(info.getPath());
-                          isDirectory = symlinkAttrs.getType().equals(FileMode.Type.DIRECTORY);
-                        } catch (IOException ifSymlinkIsBroken) {
-                          Log.w(
-                              TAG,
-                              String.format(
-                                  "Symbolic link %s is broken, skipping", info.getPath()));
-                          continue;
-                        }
+                      boolean isDirectory = false;
+                      try {
+                        isDirectory = SshClientUtils.isDirectory(client, info);
+                      } catch (IOException ifBrokenSymlink) {
+                        Log.w(TAG, "IOException checking isDirectory(): " + info.getPath());
+                        continue;
                       }
-                      HybridFileParcelable f =
-                          new HybridFileParcelable(String.format("%s/%s", path, info.getName()));
-                      f.setName(info.getName());
-                      f.setMode(OpenMode.SFTP);
-                      f.setDirectory(isDirectory);
-                      f.setDate(info.getAttributes().getMtime() * 1000);
```

Comment:
Not setting date/size/mode? in hybridFile?

Reply:
Should be fine using HybridFileParcelable constructor for SFTP files.

Reply:
Ahh, I overlooked, sorry my bad.